appengine/src/main/java/com/google/googleinterns/gscribe/resources/ExamResource.java

```diff
@@ -16,61 +16,263 @@
 
 package com.google.googleinterns.gscribe.resources;
 
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.googleinterns.gscribe.dao.*;
+import com.google.googleinterns.gscribe.models.*;
+import com.google.googleinterns.gscribe.resources.io.exception.ExamFormatException;
+import com.google.googleinterns.gscribe.resources.io.exception.InvalidDatabaseDataException;
+import com.google.googleinterns.gscribe.resources.io.exception.InvalidRequestException;
+import com.google.googleinterns.gscribe.resources.io.request.ExamInstanceRequest;
 import com.google.googleinterns.gscribe.resources.io.request.ExamRequest;
+import com.google.googleinterns.gscribe.resources.io.request.ExamSubmitRequest;
+import com.google.googleinterns.gscribe.resources.io.response.ExamInstanceResponse;
 import com.google.googleinterns.gscribe.resources.io.response.ExamResponse;
+import com.google.googleinterns.gscribe.resources.io.response.ExamSubmitResponse;
 import com.google.googleinterns.gscribe.resources.io.response.ExamsListResponse;
+import com.google.googleinterns.gscribe.services.ExamService;
+import com.google.googleinterns.gscribe.services.SpreadsheetService;
+import com.google.googleinterns.gscribe.services.TokenService;
+import com.google.inject.Inject;
 
+import javax.validation.constraints.NotNull;
 import javax.ws.rs.*;
+import java.io.IOException;
+import java.security.GeneralSecurityException;
 
 @Path("/exam")
 @Produces("application/json")
 public class ExamResource {
 
+    private final ExamService examService;
+    private final TokenService tokenService;
+    private final UserTokenDao userTokenDao;
+    private final ExamMetadataDao examMetadataDao;
+    private final QuestionsDao questionsDao;
+    private final ExamInstanceDao examInstanceDao;
+    private final AnswerDao answerDao;
+    private final SpreadsheetService spreadsheetService;
+    private final ObjectMapper objectMapper;
+
+    @Inject
+    public ExamResource(ExamService examService, TokenService tokenService, UserTokenDao userTokenDao, ExamMetadataDao examMetadataDao, QuestionsDao questionsDao, ExamInstanceDao examInstanceDao, AnswerDao answerDao, SpreadsheetService spreadsheetService, ObjectMapper objectMapper) {
+        this.examService = examService;
+        this.tokenService = tokenService;
+        this.userTokenDao = userTokenDao;
+        this.examMetadataDao = examMetadataDao;
+        this.questionsDao = questionsDao;
+        this.examInstanceDao = examInstanceDao;
+        this.answerDao = answerDao;
+        this.spreadsheetService = spreadsheetService;
+        this.objectMapper = objectMapper;
+    }
+    // TODO: Code cleanup required
```

Comment:
Be more specific, what all exact changes will be required.

Reply:
Updated.